### PR TITLE
feat(progress-bar): Add indeterminate state

### PR DESCRIPTION
**Pull request by chaitanyadeorukhkar**
- Adds indeterminate state to the progress bar
- Adds tests 
- Adds story

## Demo

**Web**

https://user-images.githubusercontent.com/24487274/208831130-97eacc94-4ac4-4f5c-a484-d0090d085e9f.mov

**iOS**

https://user-images.githubusercontent.com/24487274/208831181-dd19afd7-b111-4353-8815-a27ce66d9707.mov

**Android**

https://user-images.githubusercontent.com/24487274/209274276-1d73e58b-626b-4981-9416-e10f089d9919.mov

### CODE REVIEW

File: packages/blade/src/components/ProgressBar/ProgressBarFilled.web.tsx [hunk 1]
```diff
@@ -16,23 +17,87 @@ const pulse = keyframes`
...
+  background-color: white;
+  animation: ${pulse} ${duration} ${easing} infinite;
+  animation-delay: ${delay};
+`;
+
+const BoxWithFilledIndeterminateTransition = styled(Box)<
+  Pick<ProgressBarFilledProps, 'fillMotionDuration'>
+>(({ theme, fillMotionDuration }) => {
+  const duration = castWebType(makeMotionTime(getIn(theme.motion, fillMotionDuration)));
+  const easing = 'linear'; // TODO: Add this in motion tokens
```
Comment (chaitanyadeorukhkar) [inline packages/blade/src/components/ProgressBar/ProgressBarFilled.web.tsx:55]: We will add this to our motion tokens later (not in this PR) once design adds this too

**Review by divyanshu013 — COMMENTED** [review]
On the android demo, it seems the animation is slightly stuttering. Not sure if it's a recording issue or actual occurrence, could you please confirm?

**Comment by chaitanyadeorukhkar** [comment]
> On the android demo, it seems the animation is slightly stuttering. Not sure if it's a recording issue or actual occurrence, could you please confirm?

Looks like an emulator issue. I restarted the emulator and reinstalled blade and ran a perf on the ProgressBar. Its steady at 60fps


https://user-images.githubusercontent.com/24487274/209274276-1d73e58b-626b-4981-9416-e10f089d9919.mov

File: packages/blade/src/components/ProgressBar/ProgressBarFilled.web.tsx (hunk 1 above)
Comment (chaitanyadeorukhkar) [reply packages/blade/src/components/ProgressBar/ProgressBarFilled.web.tsx:57]: https://github.com/razorpay/blade/issues/924

File: packages/blade/src/components/ProgressBar/ProgressBar.tsx [hunk 2]
```diff
@@ -146,29 +147,40 @@ const ProgressBar = ({
...
-          valueNow: isMeter ? progressValue : percentageProgressValue,
-          valueText: isMeter ? `${progressValue}` : `${percentageProgressValue}%`,
-          valueMin: min,
-          valueMax: max,
+          valueNow: isIndeterminate ? undefined : isMeter ? progressValue : percentageProgressValue,
+          valueText: isIndeterminate
+            ? undefined
+            : isMeter
+            ? `${progressValue}`
+            : `${percentageProgressValue}%`,
```
[the remark this answers is not in the document]
Comment (chaitanyadeorukhkar) [reply packages/blade/src/components/ProgressBar/ProgressBar.tsx]: Simplified it a bit to make it more readable. See if it makes sense. 
https://github.com/razorpay/blade/pull/919/commits/b31db3248c32a06d1e366ba2b9cd608964362d0f

**Review by divyanshu013 — APPROVED** [review]
Looks good to me.

File: packages/blade/src/components/ProgressBar/ProgressBar.tsx [hunk 3]
```diff
@@ -123,7 +123,36 @@ const ProgressBar = ({
   const hasLabel = label && label.trim()?.length > 0;
   const isMeter = variant === 'meter';
   const progressValue = clamp(value, min, max);
-  const percentageProgressValue = ((progressValue - min) * 100) / (max - min);
+  const percentageProgressValue = Math.floor(((progressValue - min) * 100) / (max - min));
+  const shouldShowPercentage = showPercentage && !isMeter && !isIndeterminate;
+  const accessibilityProps: {
```
Comment (anuraghazra) [inline packages/blade/src/components/ProgressBar/ProgressBar.tsx:128]: Let pick from the AccessibilityProps if possible: 

```ts
Pick<AccessibilityProps, 'label' | 'valueMax' | 'valueNow' | 'valueMin' | 'valueText'>;
```

Comment (chaitanyadeorukhkar) [reply packages/blade/src/components/ProgressBar/ProgressBar.tsx]: Makes sense

File: packages/blade/src/components/ProgressBar/progressBarTokens.ts [hunk 4]
```diff
@@ -0,0 +1,18 @@
+const indeterminateAnimation = {
+  scaleXInitial: 1,
+  scaleXMid: 5,
+  scaleXFinal: 1,
+  leftInitial: '-8%',
+  leftMid: '50%',
+  leftFinal: '103%',
+  fillWidth: '5%',
```
Comment (anuraghazra) [inline packages/blade/src/components/ProgressBar/progressBarTokens.ts:8]: Are these values arbitary or comes from the motion design?

**Review by anuraghazra — APPROVED** [review]
Mostly looks good.

File: packages/blade/src/components/ProgressBar/progressBarTokens.ts (hunk 4 above)
Comment (chaitanyadeorukhkar) [reply packages/blade/src/components/ProgressBar/progressBarTokens.ts:8]: They're from motion design. We need it to go precisely -8% to precisely 103%

**Pull request merged**